src/compiler/scala/tools/nsc/transform/CleanUp.scala

```diff
@@ -39,6 +39,9 @@ abstract class CleanUp extends Statics with Transform with ast.TreeDSL {
     private val newStaticInits        = mutable.Buffer.empty[Tree]
     private val symbolsStoredAsStatic = mutable.Map.empty[String, Symbol]
     private var transformListApplyLimit = 8
+    private val listApply = List_apply
+    private val seqApply = Seq_apply
+    private val iseqApply = ISeq_apply
```

Comment:
This doesn't work as intended because the objects `collection.Seq` and `collection.immutable.Seq` both don't override `apply`, but just inherit it from `GenericCompanion`, which also applies for example to `collection.immutable.Set.apply`. So all of these resolve to the same symbol.

Pushed a fix proposal

Reply:
Thanks - that makes sense. I expected to see a `Set` where there should be a `Seq` but when I didn't find that I didn't know where to look